crates/api_common/src/utils.rs

```diff
@@ -551,10 +551,11 @@ pub async fn purge_post_images(
 
 /// Delete a local_user's images
 async fn delete_local_user_images(person_id: PersonId, context: &LemmyContext) -> LemmyResult<()> {
+  // This is somewhat pointless, since local_image only stores person_id. But it makes
+  // sure they're local, and their images have local rows.
   if let Ok(local_user) = LocalUserView::read_person(&mut context.pool(), person_id).await {
     let pictrs_uploads =
-      LocalImageView::get_all_by_local_user_id(&mut context.pool(), local_user.local_user.id)
-        .await?;
+      LocalImageView::get_all_by_person_id(&mut context.pool(), local_user.person.id).await?;
```

Comment:
@Nothing4You I'm not sure if this is correct anymore (since one of the points of this PR should be to allow the deletion of local thumbnails generated by federated users). 

But I'm afraid of touching it, because of the problems with deleting federated images we had before.

Reply:
This is called when an account gets banned with remove content or gets purged, so it should also delete images for remote users. Which means you can remove the whole `if let Ok(local_user) = LocalUserView::read_person` condition.

Reply:
I'm just afraid this will let ppl delete federated images again, so I'd like @Nothing4You to review before I remove this.

Reply:
Alright I'll remove, but nothing4u should look before merging.